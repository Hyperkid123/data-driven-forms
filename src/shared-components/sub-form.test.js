import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import ComponentType, { configureContext } from '../renderer-context';
import SubForm from './sub-form';

describe('<SubForm />', () => {
  it('should render with title and description', () => {
    const wrapper = mount(
      <ComponentType.Provider value={ configureContext({ formType: 'pf3' }) }>
        <SubForm title="Foo" description="Bar" fields={ [] } renderForm={ () => <div>Foo</div> } />
      </ComponentType.Provider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render with title', () => {
    const wrapper = mount(
      <ComponentType.Provider value={ configureContext({ formType: 'pf3' }) }>
        <SubForm title="Foo" fields={ [] } renderForm={ () => <div>Foo</div> } />
      </ComponentType.Provider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render with description', () => {
    const wrapper = mount(
      <ComponentType.Provider value={ configureContext({ formType: 'pf3' }) }>
        <SubForm description="Bar" fields={ [] } renderForm={ () => <div>Foo</div> } />
      </ComponentType.Provider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render only with fields', () => {
    const wrapper = mount(
      <ComponentType.Provider value={ configureContext({ formType: 'pf3' }) }>
        <SubForm fields={ [] } renderForm={ () => <div>Foo</div> } />
      </ComponentType.Provider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
