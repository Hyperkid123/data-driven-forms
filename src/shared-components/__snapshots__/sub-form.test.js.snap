// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubForm /> should render only with fields 1`] = `
<SubForm
  fields={Array []}
  renderForm={[Function]}
>
  <Col
    bsClass="col"
    componentClass="div"
    xs={12}
  >
    <div
      className="col-xs-12"
    >
      <div>
        Foo
      </div>
    </div>
  </Col>
</SubForm>
`;

exports[`<SubForm /> should render with description 1`] = `
<SubForm
  description="Bar"
  fields={Array []}
  renderForm={[Function]}
>
  <Col
    bsClass="col"
    componentClass="div"
    xs={12}
  >
    <div
      className="col-xs-12"
    >
      <p>
        Bar
      </p>
      <div>
        Foo
      </div>
    </div>
  </Col>
</SubForm>
`;

exports[`<SubForm /> should render with title 1`] = `
<SubForm
  fields={Array []}
  renderForm={[Function]}
  title="Foo"
>
  <Col
    bsClass="col"
    componentClass="div"
    xs={12}
  >
    <div
      className="col-xs-12"
    >
      <h3>
        Foo
      </h3>
      <div>
        Foo
      </div>
    </div>
  </Col>
</SubForm>
`;

exports[`<SubForm /> should render with title and description 1`] = `
<SubForm
  description="Bar"
  fields={Array []}
  renderForm={[Function]}
  title="Foo"
>
  <Col
    bsClass="col"
    componentClass="div"
    xs={12}
  >
    <div
      className="col-xs-12"
    >
      <h3>
        Foo
      </h3>
      <p>
        Bar
      </p>
      <div>
        Foo
      </div>
    </div>
  </Col>
</SubForm>
`;
